cmd/cloud.go

```diff
@@ -211,20 +211,34 @@ This will execute the test on the Load Impact cloud service. Use "k6 login cloud
 
 		var progressErr error
 		testProgress := &cloud.TestProgressResponse{}
-		percentageFmt := "[" + pb.GetFixedLengthFloatFormat(100, 2) + "%%] %s"
+		percentageFmt := "[" + pb.GetFixedLengthFloatFormat(100, 2) + "%%]"
 		progressBar.Modify(
 			pb.WithProgress(func() (float64, []string) {
 				if testProgress.RunStatus < lib.RunStatusRunning {
 					return 0, []string{testProgress.RunStatusText}
 				}
 				return testProgress.Progress, []string{
-					fmt.Sprintf(percentageFmt, testProgress.Progress*100, testProgress.RunStatusText)}
+					fmt.Sprintf(percentageFmt, testProgress.Progress*100)}
 			}),
 		)
 
 		ticker := time.NewTicker(time.Millisecond * 2000)
 		shouldExitLoop := false
 
+		var (
+			startTime   time.Time
+			maxDuration time.Duration
+		)
+		if derivedConf.Duration.Valid {
+			maxDuration = time.Duration(derivedConf.Duration.Duration)
+		} else if derivedConf.Stages != nil {
+			maxDuration = lib.SumStagesDuration(derivedConf.Stages)
+		} else if derivedConf.Execution != nil {
+			et, _ := lib.NewExecutionTuple(nil, nil)
+			execReqs := derivedConf.Execution.GetFullExecutionRequirements(et)
+			maxDuration, _ = lib.GetEndOffset(execReqs)
+		}
+
```

Comment:
Given that we always have `Execution` (or don't we) is this necessary? Also .. given that we will rename and move stuff soon, moving stuff back just so you can skip 4 lines seems unnecessary ....

Reply:
> Given that we always have Execution (or don't we) is this necessary?

True, I forgot that by this point we generate it by default anyway. I tested with just the `Execution` block, but since it takes the default `gracefulStop` of `30s` into account, the duration shown would be confusing to the user if they only specify the global `duration` or `stages` (it would show with an additional 30s). So I think handling the other cases makes sense to avoid this.

As for moving `SumStagesDuration` to lib, yeah, I'd like to avoid it because of the `Stage` renames, but umm wouldn't want to duplicate yet more code :D I don't think this will impact the renaming effort, but I'm fine with merging this afer that is done. Or just duplicating it, if you strongly prefer that. Though having identical `lib.Stage` and `executor.Stage` doesn't make much sense to me.

Reply:
Ah, sorry for the mostly duplicate comments, I missed this before making my review :man_facepalming: 

> but since it takes the default gracefulStop of 30s into account, the duration shown would be confusing to the user if they only specify the global duration or stages (it would show with an additional 30s). 

hmm maybe we should still print the same description we print in `k6 run` then? this will make it clearer what the total script time is, including `gracefulStop`... since right now, you'd show inconsistent times between when users use the shortcut options and when they use `scenarios`, despite k6 running the exact same test run... 

> Though having identical lib.Stage and executor.Stage doesn't make much sense to me.

As I mention in my review comment, they are currently not quite identical (`lib.Stage` has some cruft we don't need in the executors) and, in the future, we'd probably want to extend _only_ `executor.Stage` with extra features like non-linear ramping.

Reply:
> hmm maybe we should still print the same description we print in k6 run then?

I tried unifying the way this is printed for both `cloud` and `local` execution, see 1c7f2ce5. It's not ideal, we need better integration between local/cloud execution, and now `ExecutionRequirements` are calculated twice, though I could just pass the `maxVUs` and `maxDuration` to `printExecutionDescription()` to avoid it. Let me know.